File: 0118-pascals-triangle/0118-pascals-triangle.kt
class Solution {
    fun generate(numRows: Int): List<List<Int>> {
        val result = mutableListOf(listOf(1))

        repeat(numRows - 1) {
            result += buildList<Int> {
                add(1)
                result.last()
                    .asSequence()
                    .zipWithNext()
                    .forEach { (a, b) -> add(a + b) }
                add(1)
            }
        }

        return result
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int n) {
        vector<vector<int>> res;
        res.push_back({1});
        if(n == 1) return res;

        for(int i = 1 ; i < n; i++){
            vector<int>temp(i+1,1);
            for(int j = 1 ; j < i ; j++){
                cout<<i<<" "<<j<<endl;
               int temp2 = res[i-1][j-1]+res[i-1][j];
               temp[j] = temp2;
            }
            res.push_back(temp);

        }
        return res;
        
    }
};